frontend/src/components/nlp/components/NlpValue.tsx

```diff
@@ -64,37 +64,41 @@ export const NlpValues = ({ entityId }: { entityId: string }) => {
       params: searchPayload,
     },
   );
-  const { mutateAsync: deleteNlpValue } = useDelete(EntityType.NLP_VALUE, {
-    onError: () => {
-      toast.error(t("message.internal_server_error"));
+  const options = {
+    onError: (error: Error) => {
+      toast.error(error.message || t("message.internal_server_error"));
     },
     onSuccess() {
-      deleteEntityDialogCtl.closeDialog();
-      toast.success(t("message.item_delete_success"));
-      refetchEntity();
-    },
-  });
-  const { mutateAsync: deleteNlpValues } = useDeleteMany(EntityType.NLP_VALUE, {
-    onError: (error) => {
-      toast.error(error);
-    },
-    onSuccess: () => {
-      deleteEntityDialogCtl.closeDialog();
-      setSelectedNlpValues([]);
       toast.success(t("message.item_delete_success"));
     },
-  });
+  };
+  const { mutateAsync: deleteNlpValue } = useDelete(
```

Comment:
Are we able to use mutate instead ?
```suggestion
  const { mutate: deleteNlpValue } = useDelete(
```

Reply:
Thank you for the idea.
Yes it's possible to refactor the code and use the onSuccess callback function with a synchronous code